### ISSUE TITLE
web: Games in iframes can become unplayable if focus is lost

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#1704]
Only happens to games that require keyboard interaction.

Attached a sample: [iframegame.zip](https://github.com/ruffle-rs/ruffle/files/5597640/iframegame.zip)

- Run `index.html` (autoplay should be disabled, so you have to click the Ruffle's button).
- Start a new game.
- Press left and right arrows and spacebar: Sonic will move as expected.
- Click anywhere on the page but the game canvas.
- Click the game canvas.
- Press left and right arrows and spacebar: the page will scroll.

(If autoplay is enabled, then the game will never be playable as the iframe won't receive the focus.)

This issue affects Newgrounds:
https://www.newgrounds.com/portal/view/151706/format/flash?emulate=flash

**Comment by relrelb** [comment]
I was able to reproduce. Few notes:
* Leaving `iframe.html` with only a `keydown` event listener seems to work (the event is fired even after clicking outside the `iframe` then inside it), so this bug has something to do with Ruffle.
* I found a similar case described in https://github.com/emscripten-core/emscripten/issues/5796, and a workaround there is to call `window.focus()` after focus is lost. Pasting the following in DevTools restores the focus for me:
```js
document.getElementsByTagName("iframe")[0].contentWindow.focus();
```
The missing part is why Ruffle loses the focus and how to avoid it...
